core/runtime/src/main/java/io/quarkus/runtime/ExecutorRecorder.java

```diff
@@ -123,9 +124,11 @@ public void run() {
         };
     }
 
-    private static EnhancedQueueExecutor createExecutor(ThreadPoolConfig threadPoolConfig) {
-        final JBossThreadFactory threadFactory = new JBossThreadFactory(new ThreadGroup("executor"), Boolean.TRUE, null,
-                "executor-thread-%t", JBossExecutors.loggingExceptionHandler("org.jboss.executor.uncaught"), null);
+    private static EnhancedQueueExecutor createExecutor(ThreadPoolConfig threadPoolConfig, ThreadFactory threadFactory) {
+        if (threadFactory == null) {
+            threadFactory = new JBossThreadFactory(new ThreadGroup("executor"), Boolean.TRUE, null,
```

Comment:
So in which cases will this be `null` then?

I'm asking because for SR-CP I have a long-waiting branch that required JBossThreadFactory to let us create our own JBossThread subtype to have custom storage for ThreadLocals.

Reply:
I guess it would be `null` if there is no Vert.x thread factory.  I assume the Vert.x thread factory also has the characteristics of the Netty thread factory BTW.

But I agree that the default should probably be to use the Netty factory or at least one with fast threadlocal support.

Reply:
Right. If the `vertx-core` extension is not present, then there will be no `ThreadFactory` present.